spark/src/main/java/com/netflix/iceberg/spark/source/IcebergSource.java

```diff
@@ -109,10 +113,19 @@ protected SparkSession lazySparkSession() {
     return lazySpark;
   }
 
-  protected Configuration lazyConf() {
+  protected Configuration lazyBaseConf() {
     if (lazyConf == null) {
       this.lazyConf = lazySparkSession().sparkContext().hadoopConfiguration();
     }
     return lazyConf;
   }
+
+  protected Configuration mergeIcebergHadoopConfs(Configuration baseConf, Map<String, String> options) {
+    Configuration resolvedConf = new Configuration(baseConf);
```

Comment:
It is correct to use `new Configuration`, but that's expensive. Since this is used twice, can it copy the base config just once? Or is that complicated by making this `protected` and overriding?

Reply:
Hm, that should be fine, but I'm curious as to how expensive this has been in practice?